matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/MXOlmDevice.kt

```diff
@@ -119,7 +119,7 @@ internal class MXOlmDevice @Inject constructor(
     /**
      * @return The current (unused, unpublished) one-time keys for this account.
      */
-    fun getOneTimeKeys(): Map<String, Map<String, String>>? {
+    @Synchronized fun getOneTimeKeys(): Map<String, Map<String, String>>? {
```

Comment:
I'm a little weary of making all these functions synchronised vs ensuring they're only called from the crypto thread, at the moment some of these are called from the main thread although https://github.com/vector-im/element-android/pull/4370 should avoid ANRs at least :crossed_fingers: 

each time we synchronise we increase the risk of a deadlock :fearful:

Reply:
Ok I revert it, I don't want to spend too much time on that as we are moving to rust.
But I wonder if it's not what is causing the strange BAD_MAC when decrypting